### PR TITLE
21214: Reduces memory usage in residuals computation through more copies, easing garbage collection

### PR DESCRIPTION
**Pull request by cademack**

### CODE REVIEW

File: howso/residuals.amlg [hunk 1]
```diff
@@ -516,9 +516,9 @@
 							;assoc of all removed features to their residual values
 							removed_features_map (assoc)
 							local_cases_map (assoc)
-							case_id (current_value 1)
+							case_id (replace (current_value 1))
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove

File: howso/residuals.amlg [hunk 2]
```diff
@@ -516,9 +516,9 @@
...
 							;map of feature -> value for all the case values
-							case_values_map (zip case_features (retrieve_from_entity (current_value 1) case_features) )
+							case_values_map (zip (replace case_features) (retrieve_from_entity (current_value 1) case_features) )
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove

File: howso/residuals.amlg [hunk 3]
```diff
@@ -532,12 +532,12 @@
 						)
 						;filter out context features based on the bitmask and also create a set of all the removed features
 						(assign (assoc
-							react_context_features (filter (lambda (get remaining_feature_flags (current_index))) features)
+							react_context_features (filter (lambda (get remaining_feature_flags (current_index))) (replace features))
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove

File: howso/residuals.amlg [hunk 4]
```diff
@@ -532,12 +532,12 @@
...
 							removed_features_map
 								(if target_residual_feature
 									(zip (list target_residual_feature))
 
-									(zip (filter (lambda (not (get remaining_feature_flags (current_index)) )) features) )
+									(zip (filter (lambda (not (get remaining_feature_flags (current_index)) )) (replace features)) )
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove

File: howso/residuals.amlg [hunk 5]
```diff
@@ -560,7 +560,7 @@
 									(compute_on_contained_entities (append
 										(if focal_case
 											(query_not_in_entity_list (list case_id focal_case))
```
Comment (howsohazard) [inline howso/residuals.amlg]: add replaces for both of these

File: howso/residuals.amlg [hunk 6]
```diff
@@ -583,19 +583,19 @@
...
-											(get hyperparam_map "featureDeviations")
-											p_parameter
-											dt_parameter
+											(replace (get hyperparam_map "featureWeights"))
+											(replace !queryDistanceTypeMap)
+											(replace query_feature_attributes_map)
+											(replace (get hyperparam_map "featureDeviations"))
+											(replace p_parameter)
+											(replace dt_parameter)
 											(if valid_weight_feature weight_feature (null))
```
Comment (howsohazard) [inline howso/residuals.amlg]: (replace weight_feature)

File: howso/residuals.amlg [hunk 7]
```diff
@@ -583,19 +583,19 @@
...
 											tie_break_random_seed
```
Comment (howsohazard) [inline howso/residuals.amlg]: (replace tie_break_random_seed)

File: howso/residuals.amlg [hunk 8]
```diff
@@ -583,19 +583,19 @@
 										)
 										time_series_filter_query
 										(query_nearest_generalized_distance
-											k_parameter
-											react_context_features
+											(replace k_parameter)
+											(replace react_context_features)
 											(unzip case_values_map react_context_features)
```
Comment (howsohazard) [inline howso/residuals.amlg]: replace... probably around this unzip too

File: howso/residuals.amlg [hunk 9]
```diff
@@ -607,8 +607,8 @@
 								(map
 									(lambda (let
 										(assoc
-											feature (current_index 1)
-											case_feature_value (get case_values_map (current_index 1))
+											feature (replace (current_index 1))
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove both replaces here

File: howso/residuals.amlg [hunk 10]
```diff
@@ -627,10 +627,10 @@
 											candidate_cases_lists
 												(if (!= (null) local_cases_map)
 													(list
-														(indices local_cases_map)
-														(values local_cases_map)
+														(replace (indices local_cases_map))
```
Comment (howsohazard) [inline howso/residuals.amlg]: remove this one for sure, I lean toward removing the rest of them here

**Review by jdbeel — APPROVED** [review]

**Review by howsohazard — APPROVED** [review]

**Pull request merged**